### PR TITLE
feature #15: added sign up page

### PR DESCRIPTION
**Pull request by vihopec**
Added:
- sign up page
- common components: TextField, Button, Typography

### CODE REVIEW

**Review by Chubatik — CHANGES_REQUESTED** [review]
Move constants API_URL and TEXT_VARIANTS to folder src/constants

**Comment by YanaGushlevskaya** [comment]
Let's move common components (TextField, Button, Typography) to 'src/shared' folder

File: src/pages/sign-up/schema.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import * as yup from 'yup'
+
+export const signUpValidationSchema = yup.object({
+  username: yup.string().min(4, 'Username should be 4 chars min!').required('Username is required'),
+  password: yup.string().min(5, 'Password should be 5 chars min!').required('Password is required'),
```
Comment (YanaGushlevskaya) [inline src/pages/sign-up/schema.ts:5]: Please, change to 8 min length (according to API requirements).

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+interface IButton {
+  className?: string
+  type?: 'button' | 'submit' | 'reset' | undefined
+  disabled?: boolean
+  children: React.ReactNode
+  primary?: boolean
+}
+
+const Button = ({ className, type = 'button', disabled, children, primary = false }: IButton) => {
```
Comment (IgorLeshchenko) [inline src/components/Button/index.tsx:14]:
```
interface Props = {
   ...
}

const Button: React.FC<Props> = ({ ... }) => {
```

File: src/App.tsx [hunk 3]
```diff
@@ -13,14 +14,19 @@ import SinglePost from './pages/single-post/SinglePost'
...
-        <Route path={Routes.LOGIN} component={Login} />
-        <Route path={Routes.SINGLE_POST} component={SinglePost} />
-        <Route path={Routes.PROFILE} component={Profile} />
-        <Route component={NotFound} />
-      </Switch>
+      <div className={classes.appContainer}>
+        <Header />
+        <div className={classes.appContainer__data}>
+          <Switch>
+            <Route exact={true} path={Routes.POSTS} component={Posts} />
```
Comment (irenlian) [inline src/App.tsx:21]:
```suggestion
            <Route exact path={Routes.POSTS} component={Posts} />
```

File: src/components/TextField/index.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import classes from './TextField.module.scss'
+
+interface ITextField {
+  className?: string
+  id: string
+  name: string
+  label: string
+  type: string
+  value: string
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
```
Comment (irenlian) [inline src/components/TextField/index.tsx:13]:
```suggestion
  onChange: React.ChangeEventHandler<HTMLInputElement>
```

File: src/components/TextField/index.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  error?: boolean | undefined | string
+  helperText?: string | undefined | boolean
```
Comment (irenlian) [inline src/components/TextField/index.tsx:15]: If you use `?` it already means `undefined` as a possible value.
I wouldn't recommend storing in a single prop string & boolean simultaneously.

File: src/components/TextField/index.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+const TextField = ({ className, id, name, label, type, value, onChange, error, helperText }: ITextField) => {
```
Comment (irenlian) [inline src/components/TextField/index.tsx:18]: I'd suggest to clarify the types for components as well.
```suggestion
const TextField: React.FC< ITextField> = ({ className, id, name, label, type, value, onChange, error, helperText }) => {
```

File: src/components/Typography/index.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import { TEXT_VARIANTS } from './consts'
+
+interface ITypography {
+  className?: string
+  variant?: TEXT_VARIANTS
+  children: React.ReactNode
+}
+
+const Typography = ({ className, variant = TEXT_VARIANTS.BODY, children }: ITypography) => {
```
Comment (irenlian) [inline src/components/Typography/index.tsx:11]:
```suggestion
interface ITypography {
  variant?: TEXT_VARIANTS
}

const TypographyReact.FC<PropsWithChildren<ITypography>> = ({ className, variant = TEXT_VARIANTS.BODY, children }) => {
```

File: src/utils/api.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { API_URL } from '../consts'
+
+export const sendPostRequest = (url: string, bodyData: unknown) => {
+  return fetch(`${API_URL}${url}`, {
```
Comment (irenlian) [inline src/utils/api.ts:4]: Why did you pick `fetch` and not `axios`?

Comment (Chubatik) [reply src/utils/api.ts:4]: API will be using axios in the next PR

File: src/pages/sign-up/schema.ts (hunk 1 above)
Comment (vihopec) [reply src/pages/sign-up/schema.ts]: Fixed

File: src/components/Button/index.tsx (hunk 2 above)
Comment (vihopec) [reply src/components/Button/index.tsx]: Fixed

File: src/App.tsx (hunk 3 above)
Comment (vihopec) [reply src/App.tsx]: Fixed

File: src/components/TextField/index.tsx (hunk 4 above)
Comment (vihopec) [reply src/components/TextField/index.tsx]: Fixed

File: src/components/TextField/index.tsx (hunk 5 above)
Comment (vihopec) [reply src/components/TextField/index.tsx]: Fixed, thanks

File: src/components/TextField/index.tsx (hunk 6 above)
Comment (vihopec) [reply src/components/TextField/index.tsx]: Fixed

File: src/components/Typography/index.tsx (hunk 7 above)
Comment (vihopec) [reply src/components/Typography/index.tsx]: Fixed

**Review by Chubatik — APPROVED** [review]

**Review by IgorLeshchenko — APPROVED** [review]

**Review by YanaGushlevskaya — APPROVED** [review]

**Pull request merged**